micrometer-core/build.gradle

```diff
@@ -281,3 +281,19 @@ compileJava11TestJava {
     sourceCompatibility = JavaVersion.VERSION_11
     targetCompatibility = JavaVersion.VERSION_11
 }
+
+// jdk11 jar
+tasks.register("jdk11Jar", Jar) {
+    group = "build"
+    from(sourceSets.java11.allJava)
+    from(sourceSets.main.allJava)
```

Comment:
Hey! Projects depending on `micrometer-core` would still pull in the original jar, so classes here won't be available for them out of the box. But, end users will still be able to explicitly declare this dependency to get `jdk11` classes.

Having the same classes in both jars won't going to be a problem due to how classloaders in JVM work. I.e. there will be no duplication from the end user point of view.

I guess ideal case scenario would be to have a separate module indeed, similar to how it's done with [Kotlin stdlib](https://github.com/JetBrains/kotlin/tree/master/libraries/stdlib) for example, but that wouldn't solve the problem of people having to explicitly add `micrometer-core-jdk` dependency.

So from the end user pov, having a separate module will be more or less the same as publishing a jar with classifier here - they will have to add it explicitly anyway.
Jar with classifier just involves a bit less hassle from you in terms of maintenance. It does create some duplication from the file size pov, but given that the jar size here is <1MB, I think the Internet won't gonna break from that duplication :slightly_smiling_face:

Reply:
We can change the jar here to not include the main classes tho, then people can declare dependencies like:
```kotlin
implementation("io.micrometer:micrometer-core:1.10.0-SNAPSHOT")
implementation("io.micrometer:micrometer-core:1.10.0-SNAPSHOT:jdk11")
```

My goal was just to make it a bit simpler :slightly_smiling_face:

Reply:
I appreciate you thinking of making maintenance easier for us. It's good to consider maintenance in these kinds of things. In the end, though, I think a separate module will probably be better in this case. I worry about issues and confusion around the classifier. Would you be interested in updating your PR to make a separate micrometer-jdk11 module with the jdk11 source in micrometer-core now and deprecate the code in micrometer-core?

Reply:
Sure, can do that in a day or two :+1:

Reply:
@shakuzen I've added a new module, but task `:micrometer-java11:donwloadBaseline` now fails in CI. Not sure what should I do about it

Reply:
I can take care of that. We'll need to temporarily configure the japi check to skip the new module because it doesn't have a previous version to check against. Thanks for doing the work to get everything to this point.

Reply:
We'll need to duplicate the classes in micrometer-core and micrometer-java11 for some time. We can mark the classes in micrometer-core as deprecated so users can migrate to using the new module, but we don't want to suddenly remove them in one step. Removing them is what's causing the japi check to fail on the micrometer-core module. I'll add them back to the micrometer-core module.